### PR TITLE
Update the Babel plugin to remove empty class constructors

### LINKED ISSUE #18057: Update the Babel plugin to remove empty `class` constructors
To avoid useless `class` constructors we utilize the ESLint `no-useless-constructor` rule, which works well in *most* cases.
The exception is when a constructor includes pre-processor statements that make it conditionally empty, note e.g. how the following source code:
https://github.com/mozilla/pdf.js/blob/b6765403a1b08aa80a21cb0809741875a925ab4d/src/display/editor/tools.js#L61-L67
is transformed into (in that case for the `MOZCENTRAL` build):
https://searchfox.org/mozilla-central/rev/dbd654fa899a56a6a2e92f325c4608020e80afae/toolkit/components/pdfjs/content/build/pdf.mjs#1829

Hence it seems like a good idea to extend the Babel plugin to remove `class` constructors that become empty during building.

/cc @nicolo-ribaudo

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
> **Rename builder esprima fixtures to babel**
>  
> f724ae98a1c0859885083448e2cbc3f3d4c31cf1 changed the build
> script to use Babel, so this commit updates the file names accordingly

> **Update the Babel plugin to remove empty class constructors**
> 
> This only happens when it's safe to do so. The exceptions are:
> - when the class extends another subclass: removing the constructor would remove the error about the missing super() call
> - when there are default parameters, that could have side effects
> - when there are destructured prameters, that could have side effects

Fixes #18057

### CODE REVIEW

File: external/builder/babel-plugin-pdfjs-preprocessor.mjs [hunk 1]
```diff
@@ -234,6 +234,24 @@ function babelPluginPDFJSPreprocessor(babel, ctx) {
...
+      ClassMethod(path) {
+        const {
+          node,
+          parentPath: { parent: classNode },
+        } = path;
+        if (
+          // Remove empty constructors. We only do this for
+          // base classes, as the default constructor of derived
+          // classes is not empty (and an empty constructor
+          // must throw at runtime when constructed)
```
Comment (Snuffleupagus) [inline external/builder/babel-plugin-pdfjs-preprocessor.mjs:246]: Nit:
```suggestion
          // must throw at runtime when constructed).
```

File: external/builder/fixtures_babel/constructors.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  constructor({ x }) {}
+}
+
+class D {
+  constructor(x, y, z) {}
+}
+
+class E extends A {
+  constructor() {}
+}
```
Comment (Snuffleupagus) [inline external/builder/fixtures_babel/constructors.js:21]: Can you please also add a couple of test-cases for the situation that prompted the issue, i.e. where the pre-processor is used?
```suggestion
}

class F {
  constructor() {
    if (PDFJSDev.test('TRUE')) {
      var a = 0;
    }
  }
}

class G {
  constructor() {
    if (PDFJSDev.test('FALSE')) {
      var a = 0;
    }
  }
}
```

Comment (nicolo-ribaudo) [reply external/builder/fixtures_babel/constructors.js:21]: Done

**Review by Snuffleupagus — APPROVED** [review]
r=me, thank you!

**Pull request merged**